plat-nuvoton: configuration change

rutigl opened the pull request:
Changes load address of OPTEE-OS from 0x36000000 to 0x02100000
Implements HUK reading from DME PCR0 located in PCI mailbox
Disables DT insecure warning

jenswi-linaro commented:
Please rebase this on the upstream master branch to resolve the conflicts.

rutigl commented:
Done

> Please rebase this on the upstream master branch to resolve the conflicts.

Done

jforissier commented:
@rutigl you really need to learn how to use Git, GitHub and write proper commit subjects and descriptions. Otherwise your pull requests will take a long time to get reviewed and merged.

rutigl commented:
> @rutigl you really need to learn how to use Git, GitHub and write proper commit subjects and descriptions. Otherwise your pull requests will take a long time to get reviewed and merged.

Agree, I have not a big experience in GitHub PRs. Please tell me what to improve in current commits and I'll fix it

jforissier commented:
` plat-nuvoton: configuration change` is doing several things:
- It changes the load address of OP-TEE and the RAM size allocated to it. Why? Not explained in the commit description.
- It moves the shared memory to a different address. Why? Not explained.
- It moves the SDP memory and declares a size for it. Where was the size defined previously? was SDP used or not? does this enable SDP, or does it change its characteristics? Not explained.
- It adds `#include <mm/core_memprot.h>`. Is this neeed? If so it should be in a separate commit.
- It changes the banner text. That should be in a separate commit.

`plat-nuvoton: added HUK reading` -> `plat-nuvoton: add HUK reading` (use imperative mood in commit subject)

I had other comments but you force pushed new commits while I was reviewing so I lost them. As mentioned in the [documentation](https://optee.readthedocs.io/en/latest/general/contribute.html#review-feedback), the preferred way to address comments is to push fixup commits that are squashed in the end. And before squashing/force pushing, write a comment asking if it is OK to do so.

rutigl commented:
- Load address of OPTEE-OS was changed because we moved TF-A from SRAM to DRAM, and our new memory map is described in the TF-A code. Do you want me to put the whole new memory map in commit message or in the platform_config.h? Or enough just say the reason of moving in commit message?
- Shared memory also moved because the same reason.
- SDP size wasn't define before, it's a part of new memory map
- memprot.h include is necessary for reading HUK, because it uses phys_to_virt function to access PCI Mailbox memory
- banner text change is not necessary and I removed it
- as for style, I'll immediately change it
Thank you.

jforissier commented on core/arch/arm/plat-nuvoton/main.c:
```diff
@@ -52,3 +60,33 @@ void console_init(void)
+	void *vaddr;
+	TEE_Result res = TEE_SUCCESS;
+	uint8_t buf[NPCM_MEASURE_SIZE] = {0};
+	uint32_t bin[1 + HW_UNIQUE_KEY_LENGTH / sizeof(uint32_t)];
```
= { };

rutigl replied:
Done

jforissier reviewed (commented):
Comments on "plat-nuvoton: add HUK reading"

jforissier commented:
The subject of the second "plat-nuvoton: configuration change" commit should be "platf-nuvoton: force CFG_EXTERNAL_DT=n". With that it is:
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

jforissier commented:
The subject of the first "plat-nuvoton: configuration change" commit should be more precise:
"plat-nuvoton: change load address, shared memory and SDP memory".
With that:
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

rutigl commented on core/arch/arm/plat-nuvoton/main.c:
```diff
@@ -52,3 +60,33 @@ void console_init(void)
+
+TEE_Result tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
+{
+	void *vaddr;
```
Done

rutigl commented on core/arch/arm/plat-nuvoton/main.c:
```diff
@@ -52,3 +60,33 @@ void console_init(void)
+{
+	void *vaddr;
+	TEE_Result res = TEE_SUCCESS;
+	uint8_t buf[NPCM_MEASURE_SIZE] = {0};
```
Done

rutigl commented on core/arch/arm/plat-nuvoton/main.c:
```diff
@@ -52,3 +60,33 @@ void console_init(void)
+		return TEE_ERROR_SECURITY;
+	}
+
+	memcpy(buf, vaddr, NPCM_MEASURE_SIZE);
```
Agree, could be copied directly from vaddr

rutigl commented on core/arch/arm/plat-nuvoton/main.c:
```diff
@@ -52,3 +60,33 @@ void console_init(void)
+		return TEE_ERROR_SECURITY;
+	}
+
+	memcpy(&hwkey->data[0], bin, HW_UNIQUE_KEY_LENGTH);
```
agree, fixed

rutigl commented on core/arch/arm/plat-nuvoton/main.c:
```diff
@@ -52,3 +60,33 @@ void console_init(void)
+
+	memcpy(&hwkey->data[0], bin, HW_UNIQUE_KEY_LENGTH);
+
+	IMSG("HUK Initialized");
```
I didn't know, that this function is called several times, so I have changed the logic accordingly.

jenswi-linaro commented:
It looks like this is ready to be merged.

jforissier commented:
> It looks like this is ready to be merged.

Indeed!
